community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/fusion/FusionIndexAccessor.java

```diff
@@ -95,40 +93,30 @@ public void close() throws IOException
     @Override
     public IndexReader newReader()
     {
-        LazyInstanceSelector<IndexReader> readerSelector = new LazyInstanceSelector<>( new IndexReader[INSTANCE_COUNT],
-                slot -> instanceSelector.select( slot ).newReader() );
+        LazyInstanceSelector<IndexReader> readerSelector = new LazyInstanceSelector<>( slot -> instanceSelector.select( slot ).newReader() );
         return new FusionIndexReader( slotSelector, readerSelector, descriptor );
     }
 
     @Override
     public BoundedIterable<Long> newAllEntriesReader()
     {
-        BoundedIterable<Long>[] entries = instanceSelector.instancesAs( new BoundedIterable[INSTANCE_COUNT], IndexAccessor::newAllEntriesReader );
+        Iterable<BoundedIterable<Long>> entries = instanceSelector.flatMap( IndexAccessor::newAllEntriesReader );
```

Comment:
not quite related to this PR but still, do we need `BondedIterable<Long>` here or it can be primitive long iterator?

Reply:
Yes we could since the optimisation around maxCount never comes into play since we never only use Lucene. Let's take that in a different PR though.